services/storage/tests/unit/test_simcore_s3_dsm.py

```diff
@@ -86,3 +89,21 @@ async def _count_files(s3_file_id: SimcoreS3FileID, expected_count: int) -> None
 
     _, simcore_file_id = await upload_file(file_size, "a_file_name")
     await _copy_s3_path(simcore_file_id)
+
+
+async def test_upload_n_search(
+    simcore_s3_dsm: SimcoreS3DataManager,
+    upload_file: Callable[..., Awaitable[tuple[Path, SimcoreS3FileID]]],
+    file_size: ByteSize,
+    user_id: UserID,
+    faker: Faker,
+):
+    checksum: SHA256Str = parse_obj_as(SHA256Str, faker.sha256())
+    _, _ = await upload_file(file_size, "a_file_name", sha256_checksum=checksum)
+
+    files: list[FileMetaData] = await simcore_s3_dsm.search_owned_files(
+        user_id=user_id, file_id_prefix="", sha256_checksum=checksum
+    )
+    assert len(files) == 1
+    file: FileMetaData = files[0]
+    assert file.sha256_checksum == checksum
```

Comment:
Q: what about cehcking the file name.
Q2: what about uploading tow files with the same checksum? What is expected there?